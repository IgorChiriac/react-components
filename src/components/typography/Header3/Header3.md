```js
<Header3>Header3</Header3>
```

```js
<Header3 style={{ textAlign: 'center', color: 'salmon' }}>Header3</Header3>
```
