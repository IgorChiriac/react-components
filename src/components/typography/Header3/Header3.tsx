import React from 'react';
import styled from 'styled-components';
import * as fonts from '../../../constants/fonts';

export interface IHeader3Props extends React.HTMLAttributes<HTMLHeadingElement> {}

const SHeader = styled.h3`
  font-weight: normal;
  font-size: 24px;
  line-height: 32px;
  letter-spacing: 0;
  font-family: ${fonts.alverata};
`;

const Header3 = (props: IHeader3Props) => {
  return <SHeader {...props} />;
};

export default Header3;
