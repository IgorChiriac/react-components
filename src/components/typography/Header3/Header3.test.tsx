import { axe } from 'jest-axe';
import React from 'react';
import { render } from '@testing-library/react';
import Header3 from './Header3';

describe('<Header3 />', () => {
  it('renders the component without props with no a11y violations', async () => {
    const { container } = render(<Header3>I'm an h3 header</Header3>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
  it('renders the component with props', () => {
    const { container } = render(<Header3 style={{ textAlign: 'center' }}>I'm an h3 header</Header3>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
