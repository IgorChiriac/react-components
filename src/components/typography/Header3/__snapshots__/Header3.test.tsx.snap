// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header3 /> renders the component with props 1`] = `
.c0 {
  font-weight: normal;
  font-size: 24px;
  line-height: 32px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Alverata,"Arial Black",Verdana,sans-serif;
}

<h3
  class="c0"
  style="text-align: center;"
>
  I'm an h3 header
</h3>
`;

exports[`<Header3 /> renders the component without props with no a11y violations 1`] = `
.c0 {
  font-weight: normal;
  font-size: 24px;
  line-height: 32px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Alverata,"Arial Black",Verdana,sans-serif;
}

<h3
  class="c0"
>
  I'm an h3 header
</h3>
`;
